src/engine/Tableau.cpp

```diff
@@ -1704,6 +1704,47 @@ bool Tableau::allBoundsValid() const
     return _boundsValid;
 }
 
+// Ensure that non-basic variables are within bounds
+void Tableau::ensureNonBasicVariableGTLB( unsigned variable, double value )
+{
+    unsigned index = _variableToIndex[variable];
+    if ( !_basicVariables.exists( variable ) )
+        {
+            if ( FloatUtils::gt( value, _nonBasicAssignment[index] ) )
+                setNonBasicAssignment( variable, value, true );
+        }
+    else
+        {
+            // Recompute the status of an affected basic variable
+            // If the status changes, invalidate the cost function
+            unsigned oldStatus = _basicStatus[index];
+            computeBasicStatus( index );
+            if ( _basicStatus[index] != oldStatus )
+                _costFunctionManager->invalidateCostFunction();
+    }
+}
+
+
+// Ensure that non-basic variables are within bounds
+void Tableau::ensureNonBasicVariableLTUB( unsigned variable, double value )
+{
+    unsigned index = _variableToIndex[variable];
+    if ( !_basicVariables.exists( variable ) )
+        {
+            if ( FloatUtils::lt( value, _nonBasicAssignment[index] ) )
+                setNonBasicAssignment( variable, value, true );
+        }
+    else
+        {
+            // Recompute the status of an affected basic variable
```

Comment:
Why does this guarantee that the basic variable moves within bounds?

Reply:
I don't think that it does - the method adjusts assignment of non-basic variables and recomputes the status of basic variables. I named it for the former part, but I can name it something more inclusive - updateVariableComplianceWithLower/UpperBound? For nonBasics that means updating assignment, for basics it means updating status. What are your thoughts?

Overall this is just an encapsulation of existing code to replicate current workflow when  bound updates come through the BoundManager.

Reply:
Okay. Yes, that's fine, but please change the name to something more indicative.

Reply:
Done.